oslo/transformers/oslo_init.py

```diff
@@ -0,0 +1,451 @@
+        if isinstance(config_file_or_dict, dict):
+            # Don't modify user's data should they want to reuse it (e.g. in tests), because once we
+            # modified it, it will not be accepted here again, since `auto` values would have been overridden
+            cfg = deepcopy(config_file_or_dict)
+        elif isinstance(config_file_or_dict, str):
+            with open(config_file_or_dict, "r", encoding="utf-8") as f:
+                cfg = json.load(f)
+        else:
+            raise ValueError(
+                "expecting either a path to a oslo config file or a dict")
+        _config_check(SUPPORTED_FEATURES, cfg)
+        super(OsloTrainerConfig, self).__init__(**cfg)
+        log_dist("*** OSLO CONFIG ***")
+        if not self.is_exist('mixed_precision') or not self.mixed_precision["enable"]:
+            self.mixed_precision = None
+        else:
+            log_dist("mixed_precision: enabled")
+
+        if not self.is_exist('data_parallelism') or not self.data_parallelism["enable"]:
+            self.data_parallelism = None
+        else:
+            if self.data_parallelism['parallel_size'] is None:
+                log_dist(
+                    "data_parallelism can not be usable because parallel_size is required.",
+                    logging.WARNING
+                )
+                self.data_parallelism = None
+
+            elif self.data_parallelism['zero_stage'] is None:
+                logging.warning(
+                    "data_parallelism can not be usable because zero_stage is required."
+                )
+                self.data_parallelism = None
+            else:
+                log_dist(
+                    f"data_parallelism: enabled\n\tparallel_size: {self.data_parallelism['parallel_size']}\n\tzero_stage: {self.data_parallelism['zero_stage']}"
+                )
+                if hasattr(self.data_parallelism, 'params') and self.data_parallelism.params['cpu_offload']:
+                    self.cpu_offload = True
+
+        if not self.is_exist('sequence_parallelism') or not self.sequence_parallelism["enable"]:
+            self.sequence_parallelism = None
+        else:
+            if self.sequence_parallelism['parallel_size'] is None:
+                log_dist(
+                    "sequence_parallelism can not be usable because parallel_size is required.",
+                    logging.WARNING
+                )
+                self.sequence_parallelism = None
+            else:
+                log_dist(
+                    f"sequence_parallelism: enabled\n\tparallel_size: {self.sequence_parallelism['parallel_size']}"
+                )
+
+        if not self.is_exist('tensor_parallelism') or not self.tensor_parallelism["enable"]:
+            self.tensor_parallelism = None
+        else:
+            if self.tensor_parallelism['parallel_size'] is None:
+                log_dist(
+                    "tensor_parallelism can not be usable because parallel_size is required."
```

Comment:
how about raising ValueError rather than printing a log?

Reply:
modified

Reply:
Modified to  ValueError